bool training
bool random_agent
int64 steps
float32 discount
